### DISCUSSION TITLE
Migrate from one connection to another

### DISCUSSION THREAD

**Discussion by esalgado** [wlcrs/huawei_solar#474] [Connecting to the inverter]
Hello! I started using this integration some time ago, and I connected via the modbus and USB stick. 

I have now seen that the option to use the built in connection of the inverter to the wifi has been opened again in the firmware, so I wanted to migrate to it in order to move the raspberry pi to another place in the house.

I know I could eliminate the existing device and add it again. I have tried to not eliminate and connect a second time but it does not do it, it says "device already connected". 

The reason I do not want to remove the existing connection mode is to avoid losing all the historical data of the inverter. Do you know if there is a way to change the connection, maybe modifying the configuration yaml files?

Thanks :)

**Comment by Roving-Ronin** [comment]
@esalgado If the end result is the integration just needing to connect to a new/different IP addres, but everything else remains the same?? Then you could try going to /config/.storage/core.config_entries , search for the current configs IP address and edit it to the new IP, then restart HA (just to be sure).
No guarantees and probably worth disabling the integration whilst doing the change, then re-enabling.  Would also take a full backup of HA before, just to be safe.

**Comment by esalgado** [comment]
Hello @Roving-Ronin , 
No, I am currently connected via MODBUS and I want to connect via wifi.

I will create a virtual machine with a fresh install and connect to the inverter, then I will backup my home assistant and edit that file to replace this part with the part configured on the fresh install.

{
        "entry_id": "155a812ddab26a2f5d97ee0fdc99c22d",
        "version": 1,
        "domain": "huawei_solar",
        "title": "SUN2000-2KTL_L1",
        "data": {
          "host": null,
          "port": "/dev/serial/by-id/usb-1a86_USB_Serial-if00-port0",
          "slave_ids": [
            1
          ],
          "enable_parameter_configuration": false,
          "username": null,
          "password": null
        },
        "options": {},
        "pref_disable_new_entities": false,
        "pref_disable_polling": false,
        "source": "user",
        "unique_id": "HV2150073892",
        "disabled_by": null
      },

I will replace the "data" part with the new settings! Let's see if this works...

Thanks for the hint :)

**Comment by esalgado** [comment]
OK, got it connected to the inverter with a fresh HA virtual machine, set up the Huawei Solar, and saw that I only needed to change the host and port to ipaddress and 6607 when comparing both files. I changed it in the file in my real home assistant, restarted, and it DID NOT WORK. The configuration did not want to load... it had errors.

While I was at it I saw that the Energy panel still retained my statistics, so I removed the entry from the Huawei Solar , and added again using the web connection and port 6607...and I have just migrated from connection... it was just that easy! The power statistics just remain, no need to worry about that.

Thanks :)
